gravitee-apim-console-webui/src/management/settings/portal-ng/portal-ng.component.ts

```diff
@@ -0,0 +1,478 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, DestroyRef, OnInit, inject } from '@angular/core';
+import { FormControl, FormGroup, UntypedFormGroup } from '@angular/forms';
+import { combineLatest, Subject } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { PortalSettingsService } from '../../../services-ngx/portal-settings.service';
+import { SnackBarService } from '../../../services-ngx/snack-bar.service';
+import { PortalSettings } from '../../../entities/portal/portalSettings';
+import { GioPermissionService } from '../../../shared/components/gio-permission/gio-permission.service';
+import { CorsUtil } from '../../../shared/utils';
+
+@Component({
+  selector: 'portal-ng',
+  templateUrl: './portal-ng.component.html',
+  styleUrls: ['./portal-ng.component.scss'],
+})
+export class PortalNgComponent implements OnInit {
+  settings: PortalSettings;
+  portalForm: UntypedFormGroup;
```

Comment:
What do you think about typing this form?

Reply:
Done